setup.py

```diff
@@ -1,10 +1,23 @@
 from __future__ import with_statement
+import os
 import sys
 
 from setuptools import setup, find_packages
 from setuptools.command.test import test as TestCommand
 
-from pyoos import __version__
+def extract_version(module='pyoos'):
```

Comment:
i've seen this pattern used somewhere else - can you explain what it's accomplishing and why it's better than the simpler import version?